src/terrain/type_data.cpp

```diff
@@ -151,24 +149,29 @@ std::string terrain_type_data::get_underlying_terrain_string(const t_translation
 	return str;
 }
 
-bool terrain_type_data::try_merge_terrains(const t_translation::terrain_code & terrain) {
-
-	if(tcodeToTerrain_.count(terrain) == 0) {
-		const std::map<t_translation::terrain_code, terrain_type>::const_iterator base_iter =
-			tcodeToTerrain_.find(t_translation::terrain_code(terrain.base, t_translation::NO_LAYER));
-		const std::map<t_translation::terrain_code, terrain_type>::const_iterator overlay_iter =
-			tcodeToTerrain_.find(t_translation::terrain_code(t_translation::NO_LAYER, terrain.overlay));
+tcodeToTerrain_t::const_iterator terrain_type_data::find_or_merge(t_translation::terrain_code terrain) const
+{
+	auto i = tcodeToTerrain_.find(terrain);
+	if (i != tcodeToTerrain_.end()) {
+		return i;
+	}
+	else {
+		auto base_iter    = tcodeToTerrain_.find(t_translation::terrain_code(terrain.base, t_translation::NO_LAYER));
+		auto overlay_iter = tcodeToTerrain_.find(t_translation::terrain_code(t_translation::NO_LAYER, terrain.overlay));
 
 		if(base_iter == tcodeToTerrain_.end() || overlay_iter == tcodeToTerrain_.end()) {
-			return false;
+			return tcodeToTerrain_.end();
 		}
 
 		terrain_type new_terrain(base_iter->second, overlay_iter->second);
 		terrainList_.push_back(new_terrain.number());
-		tcodeToTerrain_.emplace(new_terrain.number(), new_terrain);
-		return true;
+		return tcodeToTerrain_.emplace(new_terrain.number(), std::move(new_terrain)).first;
 	}
-	return true; // Terrain already exists, nothing to do
+}
+
+bool terrain_type_data::try_merge_terrains(const t_translation::terrain_code & terrain)
```

Comment:
There shouldn't be a space before the ampersand.

Reply:
didn't write this code.. will probably still fix.